### ISSUE TITLE
Merge with TPOT and auto_ml?

### ISSUE THREAD

**Issue by oliverangelil** [automl/auto-sklearn#485]
I'm just curious. Why separate (and awesome) projects doing essentially the same thing? Would it not be wise to merge forces? I think this auto machine learning thing could get huge. Isn't it inefficient to have separate groups of people developing very similar tools?

**Comment by rhiever** [comment]
I agree that there is value is combining the efforts of many researchers and developers on a single AutoML project. That is somewhat the approach that the DARPA D3M project is taking: They are pulling together about a dozen AutoML research groups to develop a common, general problem-solving AutoML framework. I think in the next couple years, their approach will begin to bear fruit.

I think there are two major reasons why it's better to keep separate the R&D efforts of individual AutoML research groups:

1) We are exploring different approaches to solving the AutoML problem. We all use different solution representations (that typically boil down to some sort of ML pipeline), different search and optimization techniques, and even focus on different stages of the Machine Learning analysis pipeline.

2) Along the lines of 1), we want to encourage as many groups as possible to explore as many different AutoML ideas as possible. Typically there is a pressure to pursue the same ideas---or very similar ideas---when R&D efforts are combined into a single project. I think it's a good thing that people are looking at many different ways of solving the same problem because it increases the likelihood that we will find better solutions.

**Comment by oliverangelil** [comment]
I get your rationale for keeping things separate. Ultimately though, once enough research has been done by different AutoML groups, the best ideas should be pooled into one open source project (preferably on github). The project should be fast, easy to deploy, rich in features, etc. A merged "larger" project would likely attract more users/contributors, which would further help it grow, attracting more users/contributors, etc. Right now, a Python user interested in using AutoML has a choice of a handful of equally popular projects, giving the impression that none have gained much traction (even if some have). On the other hand, for a user interested in machine learning / statistical learning, scikit-learn would be the obvious choice and possibly one reason why the project is now actively developed by dozens of people every day.

Maybe what I am thinking of is DARPA D3M. Is it open source (I could not find it on github)? Also, are the best ideas from TPOT, auto-sklearn, auto_ml being incorporated?

**Comment by rhiever** [comment]
I don't think the DARPA D3M project is that far along yet, but they do have plans to open source a unified project within the next couple years.
